codec/codec.go

```diff
@@ -18,6 +19,24 @@ type (
 	Codec interface {
 		BinaryCodec
 		JSONCodec
+
+		// InterfaceRegistry returns the interface registry.
+		InterfaceRegistry() types.InterfaceRegistry
+
+		// GetMsgAnySigners returns the signers of the given message encoded in a protobuf Any
+		// as well as the decoded google.golang.org/protobuf/proto.Message that was used to
+		// extract the signers so that this can be used in other contexts.
+		GetMsgAnySigners(msg *types.Any) ([]string, protov2.Message, error)
+
+		// GetMsgV2Signers returns the signers of the given message.
+		GetMsgV2Signers(msg protov2.Message) ([]string, error)
+
+		// GetMsgV1Signers returns the signers of the given message plus the
+		// decoded google.golang.org/protobuf/proto.Message that was used to extract the
+		// signers so that this can be used in other contexts.
+		GetMsgV1Signers(msg proto.Message) ([]string, protov2.Message, error)
+
+		private()
```

Comment:
Is there a reason for making these interface purposely not implementable by other implementations?

Reply:
I don't think they were ever intended to be implemented externally, but we didn't mark them as such. Marking them this way makes future additions non-breaking

Reply:
Also, a private interface can be implemented - it's just necessary to embed the one we provide first

Reply:
Renamed these methods and explained how to embed to override